pkg/cluster/operation/destroy.go

```diff
@@ -171,6 +196,77 @@ func DestroyMonitored(getter ExecutorGetter, inst spec.Instance, options *spec.M
 	return nil
 }
 
+// CleanupComponent cleanup the instances
+func CleanupComponent(getter ExecutorGetter, instances []spec.Instance, cls spec.Topology, options Options, withLog bool) error {
+	timeout := options.OptTimeout
+
+	retainDataRoles := set.NewStringSet(options.RetainDataRoles...)
+	retainDataNodes := set.NewStringSet(options.RetainDataNodes...)
+
+	for _, ins := range instances {
+		// Some data of instances will be retained
+		dataRetained := retainDataRoles.Exist(ins.ComponentName()) ||
+			retainDataNodes.Exist(ins.ID())
+
+		e := getter.Get(ins.GetHost())
+		log.Infof("Cleanup instance %s", ins.GetHost())
+
+		if err := ins.WaitForDown(e, timeout); err != nil {
```

Comment:
The `stop` operation will wait for down, do we need to wait again?

Reply:
Do we need to add a `--force` flag to skip some error and force cleanup the cluster?

Reply:
No, the `stop` operation only waits `systemctl` return, it doesn't `WaitForDown`